CONTRIBUTING.md

```diff
@@ -23,28 +23,49 @@ with the [latest beta of rclone](https://beta.rclone.org/):
 If you find a bug that you'd like to fix, or a new feature that you'd
 like to implement then please submit a pull request via GitHub.
 
-If it is a big feature then make an issue first so it can be discussed.
-
-You'll need a Go environment set up with GOPATH set.  See [the Go
-getting started docs](https://golang.org/doc/install) for more info.
+If it is a big feature, then [make an issue](https://github.com/rclone/rclone/issues) first so it can be discussed.
 
 First in your web browser press the fork button on [rclone's GitHub
 page](https://github.com/rclone/rclone).
 
-Now in your terminal
+Then [install Git](https://git-scm.com/downloads) and set your public contribution [name](https://docs.github.com/en/github/getting-started-with-github/setting-your-username-in-git) and [email](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git).
+
+Next open your terminal, change directory to your preferred folder and  initialize your local rclone project:
 
     git clone https://github.com/rclone/rclone.git
     cd rclone
     git remote rename origin upstream
+      # if you have SSH keys setup in your GitHub account:
     git remote add origin git@github.com:YOURUSER/rclone.git
-    go build
+      # otherwise:
+    git remote add origin https://github.com/YOURUSER/rclone.git
+
+Note that most of the terminal commands in the rest of this guide must be executed from the rclone folder created above.
+
+Now [install Go](https://golang.org/doc/install) and verify the [GOPATH](https://github.com/golang/go/wiki/GOPATH) is pointing to a suitable folder:
+
+    go env GOPATH
+
+Great, you can now compile and execute your own version of rclone:
+
+    go build                              
+    rclone version
+
+Finally update your forked code to the latest version of the [rclone master](https://github.com/rclone/rclone/tree/master):
+
+    git checkout master
+    git fetch upstream
+    git merge --ff-only
+    git push origin --follow-tags
```

Comment:
`--follow-tags`? I don't think contributors should need to push tags should they?

Reply:
You have a point and I think the real killer question is: Do contributors have any benefit from keeping their cloned master in GitHub up to date? After playing a bit more with GitHub my answer is “No!”, there is no point in keeping it up to date – apart from a slightly tidier profile.

So to summarize the git comments I propose a change like this: (assuming fetch+merge –ff-only)

To base you changes on the latest version of the [rclone master](https://github.com/rclone/rclone/tree/master) (upstream):

    git checkout master
    git fetch upstream
    git merge --ff-only
    git checkout my-new-feature
    git rebase master

I would then move this section just below “You may like one of the popular editors/IDE's for Go”.

And finally, I suggest rephrasing “rebase to master with `git rebase master`” to “base you code on the latest version of the upstream master”. (It is in the "Make sure to bullets" app. 15 lines below)

Thanks for the review, I hope my answers make sense 😅
(Perhaps you would like to review a new consolidated proposal?)

Reply:
Yes an update to the PR would be great.

You can do this in two ways.

1. Make the changes. Add in a new commit to the branch. Push the branch
2. Make the changes. Amend the existing commit (`git commit --amend`). Push with `--force` the branch.

Either will do. for 1. GithHub gives a consolidated view. 2. is the neatest way of doing it since you are presenting the actual commit for merging, wheras with 1. I'd squash the commit (also available in the GitHub UI) before merging.

Reply:
Sure, I will force push a squashed commit shortly; being both neat and curious to see how it affects the view/history here on GitHub.

I have changed as proposed in the above comments, expanded on using git and done some minor additions/changes to make the overall text/flow more fluent.

I hope you like the changes/additions, they will be pushed shortly.